common/luaclass.c

```diff
@@ -300,6 +300,27 @@ luaA_class_connect_signal_from_stack(lua_State *L, lua_class_t *lua_class,
                                      const char *name, int ud)
 {
     luaA_checkfunction(L, ud);
+
+    /* Duplicate the function in the stack */
+    lua_pushvalue(L, ud);
+    luaA_checkfunction(L, -1);
+    luaA_checkfunction(L, ud);
+
+    /* Create a new signal to notify there is a global connection. */
+    char buf[128];
+    a_strcpy(buf, 128, name);
+    a_strcat(buf, 128, "::connected");
+
+    /* Emit a signal to notify Lua of the global connection.
+     *
+     * This can useful during initialization where the signal needs to be
+     * artificially emitted for existing objects as soon as something connects
+     * to it
+     */
+    luaA_checkfunction(L, -1);
+    luaA_class_emit_signal(L, lua_class, buf, 1);
```

Comment:
Why is this only emitted for things that connect to a class, but not when connecting to an instance?
According to `git grep`, there are four callers of `signal_connect`:

* Connecting a signal on a class (this code here)
* Connecting a signal on an object (`luaA_object_connect_signal_from_stack`)
* Connecting a signal on the `awesome` global (`luaA_awesome_connect_signal`; don't ask me why that is duplicated code)
* Connecting a signal on the `dbus` global (`luaA_dbus_connect_signal`; this code actually only allows a single function to be connected to a signal, so it might be a good idea to just ignore that)

Reply:
This is out of scope for this pull request